crates/relayer/src/chain/ckb.rs

```diff
@@ -156,63 +195,148 @@ impl CkbChain {
                 None,
             )?;
 
+        if packed_client.maximal_slot().unpack() - packed_client.minimal_slot().unpack() + 1
+            < minimal_updates_count as u64
+        {
+            // TODO: better error type
+            if let Err(err) = self.storage.rollback_to(prev_slot_opt) {
+                return Err(err.into());
+            }
+            return Err(Error::other_error(
+                "not enough updates to create multi-client".to_owned(),
+            ));
+        }
+
+        let clients = (0..client_count)
+            .map(|i| {
+                let client = packed_client.clone();
+                let client = client.as_builder().id(i.into()).build();
+                client
+            })
+            .collect::<Vec<_>>();
+        let client_info = PackedClinetInfo::new_builder()
+            .last_id(0.into())
+            .minimal_updates_count(minimal_updates_count.into())
+            .build();
+
         let tx_assembler_address = self.tx_assembler_address()?;
-        let (tx, inputs) = self
+        let (tx, inputs, type_id) = self
             .rt
-            .block_on(self.rpc_client.assemble_updates_into_transaction(
-                &tx_assembler_address,
-                packed_client,
-                packed_proof_update,
-                &self.config.lightclient_lock_typeargs,
-                &self.config.lightclient_contract_typeargs,
-                &chain_id,
-            ))?;
+            .block_on(
+                self
+                .rpc_client
+                .assemble_create_multi_client_transaction(
+                    &tx_assembler_address,
+                    clients,
+                    client_info,
+                    &self.config.lightclient_lock_typeargs,
+                    &self.config.lightclient_contract_typeargs,
+                    packed_proof_update,
+                )
+            )?;
         self.sign_and_send_transaction(tx, inputs).map_err(|err| {
             if let Err(err) = self.storage.rollback_to(prev_slot_opt) {
                 return err.into();
             }
             err
         })?;
 
+        // TODO: Write back the type id to config.
+        tracing::info!("new type_id: {}", type_id);
+        self.config.client_type_args.type_id = Some(type_id);
```

Comment:
how to persist this config object?

Reply:
As mentioned in #223 , I'm considering three ways to record this `type_id`:
      1. Writing it back to `config.toml` is an option, but it carries some risk as changing the config file during runtime may be risky.
      2. Write it to a record in `data_dir`.
      3. Users can opt to manually record it, as it only needs to be done once.

Reply:
temporarily we can persist it manually in `config.toml` at this moment for examing the feasibility of the multi-relayer feature, it won't block our work